OpenRA.Mods.Common/Traits/Buildings/Bridge.cs

```diff
@@ -47,7 +47,18 @@ class BridgeInfo : ITraitInfo, IRulesetLoaded, Requires<HealthInfo>, Requires<Bu
 
 		public object Create(ActorInitializer init) { return new Bridge(init.Self, this); }
 
-		public void RulesetLoaded(Ruleset rules, ActorInfo ai) { DemolishWeaponInfo = rules.Weapons[DemolishWeapon.ToLowerInvariant()]; }
+		public void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			if (string.IsNullOrWhiteSpace(DemolishWeapon))
+				throw new YamlException("Invalid value found for DemolishWeapon of {0}'s Bridge trait.".F(ai.Name));
```

Comment:
IsNullOrWhiteSpace can be replaced by IsNullOrEmpty like in other cases for consistency. It can't be white space probably. As mentioned, I would prefer 'Missing' than 'Invalid' as for any invalid weapon another message is used, otherwise lgtm.